src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/PreferJaggedArraysOverMultidimensional.cs

```diff
@@ -76,33 +76,37 @@ private static void AnalyzeProperty(SymbolAnalysisContext context)
         {
             var property = (IPropertySymbol)context.Symbol;
 
-            // If its an override then don't report it as it can only be fixed in the base type.
-            if (!property.IsOverride)
+            // Bail-out when override or interface implementation as it can only be fixed in the base type.
+            if (property.IsOverride || property.IsImplementationOfAnyInterfaceMember())
             {
-                if (IsMultiDimensionalArray(property.Type))
-                {
-                    context.ReportDiagnostic(property.CreateDiagnostic(DefaultRule, property.Name));
-                }
+                return;
+            }
 
-                AnalyzeParameters(context, property.Parameters);
+            if (IsMultiDimensionalArray(property.Type))
+            {
+                context.ReportDiagnostic(property.CreateDiagnostic(DefaultRule, property.Name));
             }
+
+            AnalyzeParameters(context, property.Parameters);
         }
 
         private static void AnalyzeMethod(SymbolAnalysisContext context)
         {
             var method = (IMethodSymbol)context.Symbol;
 
-            // If its an override then don't report it as it can only be fixed in the base type.
+            // Bail-out when override or interface implementation as it can only be fixed in the base type.
             // If its a getter\setter then we will report on the property instead so skip analyzing the method.
-            if (!method.IsOverride && method.AssociatedSymbol == null)
+            if (method.IsOverride || method.IsImplementationOfAnyInterfaceMember() || method.AssociatedSymbol != null)
```

Comment:
Same here - please delay `IsImplementationOfAnyInterfaceMember` as much as possible.

Reply:
I could swap the last 2 parts of the if condition but regarding delaying more the check we fall-back to my previous comment.